### DISCUSSION TITLE
Ollama support

### DISCUSSION THREAD

**Discussion by peperunas** [helix-editor/helix#9369] [Ideas]
LLMs are more and more commonly used to provide coding support. Ollama is a great project that acts as a middleware to use local LLMs.

Adding support to it would be great.

Website: https://ollama.ai

**Comment by David-Else** [comment]
You can use https://github.com/sigoden/aichat and integrate it to some degree, especially if you are using a terminal multiplexer or a terminal with split windows.

> ## Support LLMs
> 
> -   OpenAI: gpt-3.5/gpt-4/gpt-4-vision
> -   Gemini: gemini-pro/gemini-pro-vision/gemini-ultra
> -   LocalAI: opensource LLMs and other openai-compatible LLMs
> -   Ollama: opensource LLMs
> -   Azure-OpenAI: user deployed gpt-3.5/gpt-4
> -   Ernie: ernie-bot-turbo/ernie-bot/ernie-bot-8k/ernie-bot-4
> -   Qianwen: qwen-turbo/qwen-plus/qwen-max/qwen-max-longcontext/qwen-vl-plus

**Reply by peperunas** [reply]
I meant an "LSP"-like implementation. A-la https://continue.dev

**Comment by tirithen** [comment]
This is the main feature I'm missing in helix, adding gpt support via an LSP and locally run models with ollama sounds like a great way to preserve the privacy while enabling helix to use gpt models.

**Reply by ikks** [reply]
Take a look at this one. https://github.com/leona/helix-gpt/pull/42 .  Seems that ollama is available for helix-gpt
